tests/test_memory_banks.py

```
@@ -62,6 +62,33 @@ def test_create_bank_invalid_name_raises(self):
+        """Path-based bank operations must not resolve invalid names."""
+        with tempfile.TemporaryDirectory() as tmpdir:
+            data_dir = Path(tmpdir) / "data"
+            data_dir.mkdir()
+            outside = Path(tmpdir) / "outside"
+            outside.mkdir()
+            sentinel = outside / "keep.txt"
+            sentinel.write_text("do not delete")
+
+            mgr = BankManager(data_dir)
+
+            with pytest.raises(ValueError):
```

Ruff (0.15.20)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)